virtcontainers/kata_agent.go

```diff
@@ -1111,6 +1111,15 @@ func (k *kataAgent) buildContainerRootfs(sandbox *Sandbox, c *Container, rootPat
 	return nil, nil
 }
 
+func (k *kataAgent) hasAgentDebugConsole(sandbox *Sandbox) bool {
+	for _, p := range sandbox.config.HypervisorConfig.KernelParams {
+		if p.Key == "agent.debug_console" {
```

Comment:
Just a thought... although this is perfectly correct, what if we enhanced the capabilities of this flag (`agent.debug_console=false`, `agent.debug_console=bitmask` or something)? It is a little paranoid maybe, but it might be worth guarding against that by also ensuring `p.Value == ""`?

Reply:
`agent.debug_console` is the current agent ABI. If we enhance it with `agent.debug_console=false` we break the ABI and it is not necessary. For bitmask, I would suggest just adding `agent.debug_console.bitmask`. Either case, it doesn't preserve compatibility by ensuring `p.Value == ""` here, considering `p.Value == true` case ;) We will have to update both runtime and agent if we change the ABI.